src/decode/util.rs

```diff
@@ -35,33 +34,32 @@ pub fn flush_zero_padding<R: io::BufRead>(input: &mut R) -> io::Result<bool> {
 }
 
 // A Read computing a digest on the bytes read.
-pub struct HasherRead<'a, R, H>
+pub struct DigestRead<'a, 'b, R, S>
```

Comment:
Extra `'b` lifetime is specifies that the lifetime of the mutable borrows (`'a`) is different from the lifetime of the CRC algorithm (`'b`) which `Digest` borrows immutably.